src/pydicom/pixels/decoders/base.py

```diff
@@ -1098,6 +1130,10 @@ def _as_array_encapsulated(runner: DecodeRunner, index: int | None) -> "np.ndarr
             original_nr_frames = runner.number_of_frames
             for frame in frame_generator:
                 if len(frame) == runner.frame_length(unit="bytes"):
+                    if original_bits_allocated == 1 and runner.get_option(
+                        "is_bitpacked"
+                    ):
+                        frame = unpack_bits(frame)
```

Comment:
Does this need the `[:pixel_per_frame]` range?

Reply:
Thanks, addressed in [41dd044](https://github.com/pydicom/pydicom/pull/2216/commits/41dd044bfc9ea9b5559baca88a411ff7a523781a)